nanolog_test.go

```diff
@@ -48,7 +48,7 @@ func TestSetWriter(t *testing.T) {
 	SetWriter(buf)
 
 	// simulate some logging
-	defaultLogWriter.w.WriteByte(35)
+	defaultLogWriter.(*logWriter).w.WriteByte(35)
```

Comment:
These are all very ugly and don't really add anything to the tests. They should probably just be instantiating a new LogWriter each time.